core/src/main/java/org/mule/runtime/core/internal/execution/ModuleFlowProcessingPhase.java

```diff
@@ -124,7 +125,7 @@ public void runPhase(final ModuleFlowProcessingPhaseTemplate template, final Mes
                        final PhaseResultNotifier phaseResultNotifier) {
     try {
       final MessageSource messageSource = messageProcessContext.getMessageSource();
-      final FlowConstruct flowConstruct = (FlowConstruct) muleContext.getConfigurationComponentLocator().find(Location.builder()
+      final FlowConstruct flowConstruct = (FlowConstruct) componentLocator.find(Location.builder()
```

Comment:
The fact that `muleContext.getConfigurationComponentLocator()` foes not cache the value internally is inconsistent with other muleContext getters such as `getExpressionManager` ``getLockFactory` etc.